### DISCUSSION TITLE
Send transaction in wallet

### DISCUSSION THREAD

**Discussion by fereshteh-jvz** [WalletConnect/walletconnect-monorepo#3371] [Developer Support]
I'm developing a wallet. In the documentation it says wallet should send transaction hash as a result to eth_sendTransaction request. So in wallet we should broadcast the transaction and send the transaction hash to dapp via  Web3Wallet.respondSessionRequest. But when i send the transaction hash to uni swap, it does not show any reaction. Am i doing something wrong? this is my sample code:

          val transactionHash = "0x51ad2a7bef9e87db0444cde96bb5262dda74c9f561820bbf413ef50e057c6af5"

           val response = Params.SessionRequestResponse(
               sessionTopic = sessionRequest.topic,
                jsonRpcResponse = Model.JsonRpcResponse.JsonRpcResult(
                sessionRequest.request.id,
                transactionHash
               )
              )

            Web3Wallet.respondSessionRequest(response, onSuccess = {

            }) { onError ->
           }

**Comment by mshahjahan-mufin** [comment]
@fereshteh-jvz I was stuck on the same thing. You have to load your contract abi file and call your transfer function that will generate a hex and you can send it.
You can check detailed info for how data parameter is calcualted at here: [Transaction Data hash](https://medium.com/mycrypto/why-do-we-need-transaction-data-39c922930e92)

**Reply by shouvikchy** [reply]
@mshahjahan-mufin would you please demonstrate the transfer process?i am facing the same problem.uniswap approves my swap transaction but it fails after a while.
